actors/market/src/lib.rs

```diff
@@ -787,14 +809,19 @@ impl Actor {
                         amount_slashed += slash_amount;
 
                         // Delete proposal and state simultaneously.
-                        let deleted = st.remove_deal_state(rt.store(), deal_id)?;
-
-                        if deleted.is_none() {
-                            return Err(actor_error!(
-                                illegal_state,
-                                "failed to delete deal state: does not exist"
-                            ));
-                        }
+                        let _deleted =
+                            st.remove_deal_state(rt.store(), deal_id)?.ok_or_else(|| {
```

Comment:
Revert this one too – we're only changing exit codes for proposals, not states.

Reply:
Done 👍🏾